contracts/dtoken/src/supply.rs

```diff
@@ -90,6 +119,8 @@ impl Contract {
 
         // Dtokens minting and adding them to the user account
         self.mint(self.get_signer_address(), dtoken_amount.into());
```

Comment:
@pahomas do we actually need this mint \ burn? if we rely only on total_supplies most likely we don't use `dtoken.token.total_supply` anymore

the only question is that we still use `exchange_rate` to convert withdraw amount and have it on UI but maybe we can get rid of it to simplify the flow?